runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6960,6 +6960,13 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
                   }
                }
             }
+         else
+            {
+            // When both canDoRelocatableCompile and TR::Options::canJITCompile() are false,
+            // remote compilation request should not be used.
+            doLocalCompilation = true;
+            }
+
```

Comment:
My suggestion would be to move the code below under `if (TR::Options::canJITCompile())`. Then you don't have to add this new code.